addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/BuildingMonitorHandler.java

```diff
@@ -0,0 +1,387 @@
+
+    public BuildingMonitorHandler(Thing thing) {
+        super(thing);
+        gson = new Gson();
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.debug("Initializing thing handler.");
+
+        BuildingMonitorConfiguration config = getConfigAs(BuildingMonitorConfiguration.class);
+        logger.debug("config apikey = (omitted from logging)");
+        logger.debug("config deviceId = {}", config.deviceId);
+        logger.debug("config refresh = {}", config.refresh);
+
+        boolean validConfig = true;
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.apikey) == null) {
+            errorMsg = "Parameter 'apikey' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (StringUtils.trimToNull(config.deviceId) == null) {
+            errorMsg = "Parameter 'deviceId' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (config.refresh == null) {
+            errorMsg = "Parameter 'refresh' must be configured";
+            validConfig = false;
+        } else if (config.refresh < 2) {
+            errorMsg = "Parameter 'refresh' must be at least 2 minutes";
+            validConfig = false;
+        }
+        bridgeHandler = getBridgeHandler();
+        if (bridgeHandler == null || !bridgeHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+            logger.debug("Bridge handler not found or not ONLINE.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+
+        } else {
+            BridgeConfiguration configBridge = bridgeHandler.getBridgeConfiguration();
+            if (validConfig) {
+                updateStatus(ThingStatus.OFFLINE);
+                startAutomaticRefresh(config, configBridge);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            }
+        }
+    }
+
+    protected synchronized BridgeHandler getBridgeHandler() {
+        Bridge bridge = getBridge();
+
+        if (bridge == null) {
+            logger.debug("unable to get bridge");
+            return null;
+        }
+        ThingHandler handler = bridge.getHandler();
+        if (handler instanceof BridgeHandler) {
+            return (BridgeHandler) handler;
+        } else {
+            logger.debug("unable to get bridge handler");
+            return null;
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the wso2iots handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        bridgeHandler = getBridgeHandler();
+        if (bridgeHandler == null || !bridgeHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
```

Comment:
`Enum` see above.

Reply:
@htreu I always forget do you need this ONLINE check of does ESH already do this?